### PR TITLE
CLDIDE-2537 Add ability to configure extra hosts for docker images.

### PR DESCRIPTION
**Pull request by skabashnyuk**
This set of lines will be add to /etc/hosts on running docker container

### CODE REVIEW

File: plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProviderTest.java [hunk 1]
```diff
@@ -840,7 +855,7 @@ public void shouldBindCommonAndDevVolumesToContainerOnDevInstanceCreationFromSna
     }
 
     @Test
-    public void shouldBindCommonVolumesOnlyToContainerOnNonDevInstanceCreationFromRecipe() throws Exception {
+         public void shouldBindCommonVolumesOnlyToContainerOnNonDevInstanceCreationFromRecipe() throws Exception {
```
Comment (garagatyi) [inline plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProviderTest.java]: Please reformat code here

File: plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProviderTest.java [hunk 2]
```diff
@@ -873,6 +889,77 @@ public void shouldBindCommonVolumesOnlyToContainerOnNonDevInstanceCreationFromRe
...
+
+        ArgumentCaptor<HostConfig> argumentCaptor = ArgumentCaptor.forClass(HostConfig.class);
+        verify(dockerConnector).startContainer(anyString(), argumentCaptor.capture());
+
+        final String[] extraHosts = argumentCaptor.getValue().getExtraHosts();
+        assertEquals(extraHosts.length, 2);
+        assertEquals(extraHosts[0], "dev.box.com:192.168.0.1");
+        assertEquals(extraHosts[1], "codenvy.com.com:185");
+    }
     @Test
```
Comment (garagatyi) [inline plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProviderTest.java]: Please add tests:
shouldAddExtraHostOnNonDevInstanceCreationFromRecipe
shouldAddExtraHostOnNonDevInstanceCreationFromSnapshot

**Comment by garagatyi** [comment]
LGTM

**Comment by garagatyi** [comment]
LGTM

**Comment by evoevodin** [comment]
:+1:

**Pull request merged**